app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.order(market_name: :asc)
+    @all_markets = Market.order(market_name: :asc)
+  end
+
+  def show
+    @market = Market.find(params[:id])
+    @vendors = Vendor.where(market_id: params[:id])
+    render :show
+  end
+
+  def lettermarket
+    @markets = Market.where("market_name LIKE :prefix", prefix: "#{params[:letter]}%")
+    @all_markets = Market.order(market_name: :asc)
+  end
+
+  def new
+    @market = Market.new
+  end
+
+  def create
+    @market = Market.new(market_edit_params[:market])
+    if @market.save
+      redirect_to market_path(@market.id)
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @market = Market.find(params[:id])
+  end
+
+  def update
+    @market = Market.update(params[:id],market_edit_params[:market])
+    if @market.save
+      redirect_to market_path
+    else
+      render :market
+    end
+  end
+
+  def destroy
+    @mark_id = Vendor.find(params[:id]).market_id
+    @vendor = Vendor.find(params[:id]).destroy
+    redirect_to market_path(@mark_id)
+  end
+
+  private
+  def market_edit_params
+    params.permit!
```

Comment:
This is a potentially dangerous way of allowing attributes. This is where you can let the app know that there are only specific attributes that can be passed in the params hash. When you specifically whitelist attributes, users will get an error if they try to post other, forbidden attributes.

http://blog.trackets.com/2013/08/17/strong-parameters-by-example.html